frontend/src/metabase/questions/components/QuestionActivityTimeline.jsx

```diff
@@ -63,28 +70,56 @@ export default _.compose(
   }),
   connect(
     null,
-    () => {
-      return {
-        revertToRevision,
-      };
+    {
+      revertToRevision,
     },
   ),
 )(QuestionActivityTimeline);
 
-// this looks ugly, so not showing it for now
-function renderQuestionActivityTimelineFooter(item, revertToRevision) {
-  if (item.showFooter) {
+QuestionActivityTimelineFooter.propTypes = {
+  item: PropTypes.object.isRequired,
+  onRevisionClick: PropTypes.func.isRequired,
+  onRequestClick: PropTypes.func.isRequired,
+};
+
+function QuestionActivityTimelineFooter({
+  item,
+  onRevisionClick,
+  onRequestClick,
+}) {
+  const { revision, request } = item;
+  if (request) {
     return (
-      <div className="py1 flex justify-end">
+      <div className="py1">
+        <Button
+          className={cx(
+            "p0 borderless text-underline-hover bg-transparent-hover",
+            isRequestOpen(request)
+              ? "text-dark text-dark-hover"
+              : "text-light text-light-hover",
+          )}
+          onClick={() => onRequestClick(request)}
+        >
+          {item.requestStatusText}
+        </Button>
+      </div>
+    );
+  } else if (revision) {
+    return (
+      <div className="py1">
         <ActionButton
-          actionFn={() => revertToRevision(item.revision)}
-          className="Button-borderless text-error"
-          normalText={t`Revert`}
+          actionFn={() => onRevisionClick(revision)}
+          className="p0 borderless text-accent3 text-accent3-hover text-underline-hover bg-transparent-hover"
+          successClassName=""
+          failedClassName=""
+          normalText={t`Revert back`}
           activeText={t`Reverting…`}
           failedText={t`Revert failed`}
           successText={t`Reverted`}
         />
```

Comment:
Should we maybe split this component into two?

Reply:
good idea, how's this?